server/src/main/java/org/apache/iotdb/db/query/expression/ResultColumn.java

```diff
@@ -88,12 +88,15 @@ public ResultColumn(Expression expression) {
   /**
    * @param prefixPaths prefix paths in the from clause
    * @param resultColumns used to collect the result columns
+   * @param isGroupByLevel used to skip illegal alias judgement here. Because count(*) may be *
+   *     unfolded to more than one expression, but it still can be aggregated together later
    */
-  public void concat(List<PartialPath> prefixPaths, List<ResultColumn> resultColumns)
+  public void concat(
+      List<PartialPath> prefixPaths, List<ResultColumn> resultColumns, boolean isGroupByLevel)
```

Comment:
After discussing with Steve Yurong SU @SteveYurongSu , it is not very elegant to add `isGroupByLevel` here. It' better to add an interface, like `Validator` which has a `validate` method in it. And AliasValidator implements that interface, and GroupByLevelAliasValidator `override` that `validate` method.

Reply:
It's not necessary I think....
Let's think the impelement way. Do I need to implement a validator in every ResultColumn? It's not expected obviously.
Or pass the validator as a parameter of this method? Then what's the different from now?

Reply:
We can pass the validator as a parameter of this method. It's better to extend. If we have another rules about the alias or something else, we can implement another `Validator` and passing it instead of adding another `boolean` parameter, like `isGroupByLevel`.